include/cocaine/api/isolate.hpp

```diff
@@ -67,8 +68,29 @@ struct isolate_t {
     }
 
     virtual
+    void
+    spool() = 0;
+
+    // Default implementation delegates the control flow into the blocking spool method.
+    virtual
     std::unique_ptr<cancellation_t>
-    spool(callback_type cb) = 0;
+    async_spool(callback_type cb) {
+        std::unique_ptr<cancellation_t> cancellation(new cancellation_t());
```

Comment:
It looks like exception processing is common and should not be overridden, however cancelation token creation can and should be overridden. Maybe it's better leave cancellation creation in spool? or move it out to some extra method? Do we really need both sync and async implementations for any isolation at the same time? Also async_spool is not a really good name for this method as it has default behaviour which is sync. I'd rather  make it pure virtual with implementation in this case.